### PR TITLE
Landscape Mode

### PR DESCRIPTION
**Pull request by barbarakos**
App supports landscape mode

### CODE REVIEW

File: QuizApp/Extensions/BackgroundGradient.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        colors = [
+            UIColor(red: 0.453, green: 0.308, blue: 0.637, alpha: 1).cgColor,
+            UIColor(red: 0.154, green: 0.185, blue: 0.463, alpha: 1).cgColor
+        ]
+
+        locations = [0, 1]
+        startPoint = CGPoint(x: 0.5, y: 0.25)
+        endPoint = CGPoint(x: 0.5, y: 0.85)
+        type = .axial
+    }
```
Comment (dorastipkovic) [inline QuizApp/Extensions/BackgroundGradient.swift]: set these in init

File: QuizApp/Quiz/ViewControllers/QuizDetailsViewController.swift [hunk 2]
```diff
@@ -112,21 +121,23 @@ extension QuizDetailsViewController: ConstructViewsProtocol {
 
     func defineLayoutForViews() {
         gradientLayer.frame = view.bounds
-        gradientLayer.locations = [0, 1]
+        navigationItem.titleView = titleLabel
```
Comment (dorastipkovic) [inline QuizApp/Quiz/ViewControllers/QuizDetailsViewController.swift]: Move this someplace else. in styleViews or viewDidLoad

File: QuizApp/User/ViewControllers/UserViewController.swift [hunk 3]
```diff
@@ -38,6 +38,14 @@ class UserViewController: UIViewController {
         userViewModel.getUser()
     }
 
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        guard let gradient = gradientLayer else { return }
+
+        gradient.frame = CGRect(x: 0, y: 0, width: size.width, height: size.height)
```
Comment (dorastipkovic) [inline QuizApp/User/ViewControllers/UserViewController.swift]:
```suggestion
        gradientLayer?.frame = CGRect(x: 0, y: 0, width: size.width, height: size.height)
```

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/Extensions/BackgroundGradient.swift [hunk 4]
```diff
@@ -2,6 +2,20 @@ import UIKit
...
 
+    override init() {
+        super.init()
+
+        setBackground()
+    }
+
+    override init(layer: Any) {
+        super.init(layer: layer)
+    }
```
Comment (dorastipkovic) [inline QuizApp/Extensions/BackgroundGradient.swift:13]: Is this needed?

File: QuizApp/Extensions/BackgroundGradient.swift [hunk 5]
```diff
@@ -2,6 +2,20 @@ import UIKit
...
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
     func setBackground() {
```
Comment (dorastipkovic) [inline QuizApp/Extensions/BackgroundGradient.swift]: make private

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/Extensions/BackgroundGradient.swift (hunk 4 above)
Comment (barbarakos) [reply QuizApp/Extensions/BackgroundGradient.swift:13]: The application crashes without it when the simulator is put in landscape mode.

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**